sdk/documentintelligence/Azure.AI.DocumentIntelligence/src/TrainingOperation.cs

```diff
@@ -46,38 +47,23 @@ public override async ValueTask<Response> UpdateStatusAsync(CancellationToken ca
 
         private string GetOperationId()
         {
-            const string OperationIdNotFoundErrorMessage = "The operation ID was not present in the service response.";
             var response = GetRawResponse();
 
-            // If the "operation-location" header is present, it means the latest response came from a POST request,
-            // so the operation ID can be extracted from the header.
-            // Otherwise, the latest response came from a GET request, so the operation ID can be obtained directly
-            // from the response's content.
-
             if (response.Headers.TryGetValue("operation-location", out string operationLocation))
             {
-                // "operation-location" header pattern:
-                // https://<endpoint>/documentintelligence/operations/<operationId>?api-version=<version>
-                var match = Regex.Match(operationLocation, @"[^:]+://[^/]+/documentintelligence/operations/([^?]+)");
+                // Location header pattern:
+                // https://<endpoint>/documentintelligence/<api>/<operationId>?api-version=<version>
+                var match = Regex.Match(operationLocation, @"[^:]+://[^/]+/documentintelligence/.+/([^?/]+)");
```

Comment:
You may want to consider hoisting this expression up to a static and making it precompiled.   If you don't see this operation being called multiple times, then ignore me.

Reply:
It definitely can be called many times. Done!